modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RapidExperimentationTest.java

```diff
@@ -0,0 +1,328 @@
+
+        // since the experiment crossed the maxusers the experiment should change to PUASED
+        Assignment assignment = postAssignment(exp, new User("user10"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        assignment = postAssignment(exp, new User("user11"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        assignment = postAssignment(exp, new User("user12"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        exp = getExperiment(exp);
+        Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_PAUSED);
+        expList.set(0, rapidExperiment1);
+    }
+
+    /**
+     * This test case covers a scenario where we start an experiment, do assignments and after that update the
+     * experiment to rapid experiment with max users less than the number of users already assigned
+     */
+    @Test
+    public void testChangeRegularExperimentToRapidExperiment() {
+        clearAssignmentsMetadataCache();
+        Experiment exp = getExperiment(expList.get(1));
+
+        for (int i = 1; i <= 5; i++) {
+            // lets do an assignment for a user with context set to PROD
+            Assignment assignment = postAssignment(exp, new User("user" + i), "PROD");
+
+            // lets assert the status and response code and also the state of the experiment
+            assertReturnCode(response, HttpStatus.SC_OK);
+            Assert.assertEquals(assignment.status, "NEW_ASSIGNMENT");
+            Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_RUNNING);
+        }
+
+        exp = exp.setIsRapidExperiment(true).setUserCap(3);
+        exp = putExperiment(exp);
+
+        clearAssignmentsMetadataCache();
+        System.out.println("*****" + exp);
+        // since the experiment crossed the maxusers the experiment should change to PUASED
+        Assignment assignment = postAssignment(exp, new User("user10"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        exp = getExperiment(exp);
+        Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_PAUSED);
+        expList.set(1, rapidExperiment2);
+    }
+
+    /**
+     * This test case tests for the put assignment API call that forces assignment call on particular bucket for a rapid
+     * experiment
+     */
+    @Test
+    public void testPutAssignmentExperimentToRapidExperiment() {
+        Experiment exp = getExperiment(expList.get(2));
+        List<Bucket> bucketList = getBuckets(exp);
+        Assignment assignment = AssignmentFactory.createAssignment().setAssignment(bucketList.get(0).label)
+                .setExperimentLabel(exp.label).setOverwrite(true);
+        for (int i = 1; i <= 5; i++) {
```

Comment:
@sudheerkethireddy Can we please replace the hardcoded `5` with the constant everywhere in the tests?